engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventTest.java

```diff
@@ -134,6 +137,69 @@ public void shouldNotDeployCompensationHandlerWithBoundaryEvent() {
             Task.class, "A compensation handler should have no boundary events"));
   }
 
+  @Test
+  public void shouldDeployCompensationEventSubprocess() {
+    final var process =
+        Bpmn.createExecutableProcess("compensation-process")
+            .startEvent()
+            .subProcess(
+                "subprocess",
+                s ->
+                    s.embeddedSubProcess()
+                        .eventSubProcess(
+                            "embedded-subprocess",
+                            eventSubProcess ->
+                                eventSubProcess
+                                    .startEvent(
+                                        "compensation-start",
+                                        AbstractStartEventBuilder::compensation)
+                                    .intermediateThrowEvent(
+                                        "compensation-throw-event",
+                                        AbstractThrowEventBuilder::compensateEventDefinition)
+                                    .userTask("B")
+                                    .endEvent())
+                        .startEvent()
+                        .userTask("A")
+                        .boundaryEvent()
+                        .compensation(c -> c.userTask("undo-A"))
+                        .endEvent()
+                        .done())
+            .endEvent(
+                "compensation-end-event",
+                end -> end.compensateEventDefinition().compensateEventDefinitionDone())
+            .done();
+
+    ProcessValidationUtil.validateProcess(process);
+  }
+
+  @Test
+  public void shouldNotDeployCompensationEventSubprocessOnProcessLevel() {
+    final var process =
+        Bpmn.createExecutableProcess("compensation-process")
+            .startEvent()
+            .subProcess(
+                "subprocess",
+                s ->
+                    s.embeddedSubProcess()
+                        .startEvent("compensation-start", AbstractStartEventBuilder::compensation)
+                        .intermediateThrowEvent(
+                            "compensation-throw-event",
+                            AbstractThrowEventBuilder::compensateEventDefinition)
+                        .userTask("B")
+                        .endEvent()
+                        .done())
+            .endEvent(
+                "compensation-end-event",
+                end -> end.compensateEventDefinition().compensateEventDefinitionDone())
+            .done();
+
+    ProcessValidationUtil.validateProcess(
+        process,
+        ExpectedValidationResult.expect(
+            SubProcess.class,
+            "A compensation event subprocess is not allowed on the process level"));
```

Comment:
:thinking: In this case, the validator returns also the following validation error message:

```
Start events in subprocesses must be of type none
```

Reply:
Is it not correct ?

Reply:
The failure message is correct. I want to highlight that we already verified that an embedded subprocess has only a none start event. :sweat_smile: